core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2250,6 +2250,25 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
+   * Kill a given task. It will be retried.
+   *
+   * @param taskId the task ID to kill
+   */
+  def killTask(taskId: Long): Unit = {
+    killTask(taskId, "cancelled")
+  }
+
+  /**
+   * Kill a given task. It will be retried.
+   *
+   * @param taskId the task ID to kill
+   * @param reason the reason for killing the task, which should be a short string
+   */
+  def killTask(taskId: Long, reason: String): Unit = {
```

Comment:
I am unclear about the expectations from the api.
- What is the expectation when a task is being killed.
  - Is it specifically for the task being referenced; or all attempts of the task ?
  - If latter, do we discard already succeeded tasks ?
  - "killAndRescheduleTask" implies it will be rescheduled - which might not occur in case this was a speculative task (or already completed) : would be good to clarify.
- Is this expected to be exposed via the UI ?
  - How is it to be leveraged (if not via UI) ? 
- How to get to taskId - via SparkListener.onTaskStart ?
  - Any other means ? (IIRC no other way to get at it, but good to clarify for api user)